conans/client/remote_manager.py

```diff
@@ -35,24 +35,24 @@ def upload_conan(self, conan_reference, remote, retry, retry_wait, ignore_delete
 
         if CONANFILE not in files or CONAN_MANIFEST not in files:
             raise ConanException("Cannot upload corrupted recipe '%s'" % str(conan_reference))
-
         the_files = compress_recipe_files(files, export_folder, self._output)
-
-        ret = self._call_remote(remote, "upload_conan", conan_reference, the_files,
-                                retry, retry_wait, ignore_deleted_file)
-        duration = time.time() - t1
-        log_recipe_upload(conan_reference, duration, the_files, remote)
-        msg = "Uploaded conan recipe '%s' to '%s'" % (str(conan_reference), remote.name)
-        # FIXME: server dependent
-        if remote.url == "https://server.conan.io":
-            msg += ": https://www.conan.io/source/%s" % "/".join(conan_reference)
+        if not dry_run:
+            ret = self._call_remote(remote, "upload_conan", conan_reference, the_files,
+                                    retry, retry_wait, ignore_deleted_file)
+            duration = time.time() - t1
+            log_recipe_upload(conan_reference, duration, the_files, remote)
+            msg = "Uploaded conan recipe '%s' to '%s'" % (str(conan_reference), remote.name)
+            # FIXME: server dependent
+            if remote.url == "https://server.conan.io":
+                msg += ": https://www.conan.io/source/%s" % "/".join(conan_reference)
+            else:
+                msg += ": %s" % remote.url
+            self._output.info(msg)
+            return ret
         else:
-            msg += ": %s" % remote.url
-        self._output.info(msg)
+            return None
 
-        return ret
-
-    def upload_package(self, package_reference, remote, retry, retry_wait):
+    def upload_package(self, package_reference, remote, retry, retry_wait, dry_run = None):
```

Comment:
For default params, the style is without spaces, like ``dry_run=None``. Also, if it is a boolean, you might want to use ``False`` as default